src/Core/Managed/Shared/Extensibility/Implementation/SimpleMetricStatisticsAggregator.cs

```diff
@@ -0,0 +1,116 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Threading;
+
+    /// <summary>
+    /// Represents mechanism to calculate basic statistical parameters of a series of numeric values.
+    /// </summary>
+    internal class SimpleMetricStatisticsAggregator
+    {
```

Comment:
I have note yet read the whole CR so I may learn this shortly (sorry if that's the case), but what do you use these getters for and how important is it that they give consistent values? You may want to put them under lock too.